src/bun.js/bindings/bindings.zig

```diff
@@ -445,77 +447,66 @@ pub const ZigString = extern struct {
 
     pub fn toSliceFast(this: ZigString, allocator: std.mem.Allocator) Slice {
         if (this.len == 0)
-            return Slice{ .ptr = "", .len = 0, .allocator = allocator, .allocated = false };
+            return Slice.empty;
         if (is16Bit(&this)) {
             var buffer = this.toOwnedSlice(allocator) catch unreachable;
             return Slice{
                 .ptr = buffer.ptr,
                 .len = @truncate(u32, buffer.len),
-                .allocated = true,
-                .allocator = allocator,
+                .allocator = NullableAllocator.new(allocator),
             };
         }
 
         return Slice{
             .ptr = untagged(this.ptr),
             .len = @truncate(u32, this.len),
-            .allocated = false,
-            .allocator = allocator,
+            .allocator = NullableAllocator{},
         };
     }
 
     /// This function checks if the input is latin1 non-ascii
     /// It is slow but safer when the input is from JavaScript
     pub fn toSlice(this: ZigString, allocator: std.mem.Allocator) Slice {
         if (this.len == 0)
-            return Slice{ .ptr = "", .len = 0, .allocator = allocator, .allocated = false };
+            return Slice{ .ptr = "", .len = 0, .allocator = NullableAllocator{} };
         if (is16Bit(&this)) {
             var buffer = this.toOwnedSlice(allocator) catch unreachable;
             return Slice{
+                .allocator = NullableAllocator.new(allocator),
                 .ptr = buffer.ptr,
                 .len = @truncate(u32, buffer.len),
-                .allocated = true,
-                .allocator = allocator,
             };
         }
 
         if (!this.isUTF8() and !strings.isAllASCII(untagged(this.ptr)[0..this.len])) {
             var buffer = this.toOwnedSlice(allocator) catch unreachable;
-            return Slice{
-                .ptr = buffer.ptr,
-                .len = @truncate(u32, buffer.len),
-                .allocated = true,
-                .allocator = allocator,
-            };
+            return Slice.fromUTF8(buffer);
         }
 
         return Slice{
             .ptr = untagged(this.ptr),
             .len = @truncate(u32, this.len),
-            .allocated = false,
-            .allocator = allocator,
+            .allocator = NullableAllocator,
```

Comment:
Is this right? The type name instead of the value?

Reply:
Ah no, Zig wasn't checking that code because the function wasn't used.